lib/array_equals.rb

```diff
@@ -1,5 +1,25 @@
-# Determines if the two input arrays have the same count of elements
-# and the same integer values in the same exact order
 def array_equals(array1, array2)
+  if array1 == nil || array2 == nil || array1 == [] || array2 == []
+    if array1 == array2
+      return true
+    else
+      return false
+    end
+  end
+
+  if array1.length != array2.length
+    return false
+  else
```

Comment:
You don't need this `else` because you will get to this line assuming line 11 does not return false.